lib/webidl2.js

```diff
@@ -58,6 +58,15 @@ function parseByTokens(tokeniser, options) {
   }
 
   function definition() {
+    if (options.productions) {
+      for (const production of options.productions) {
```

Comment:
Q: maybe silly question, as I'm not 100% on how `productions` work, but seems odd that this would run through just one production, and then return after the first result is gotten? Is that correct?

Reply:
It must not be possible to match multiple productions at once, since it makes the grammar ambiguous. But here it does not actually check that and just assumes the first match is the unique match.